docs/community/contribute/checks/overview.md

```diff
@@ -0,0 +1,122 @@
+    - `severity` is the severity of the check. This should be one of `LOW`, `MEDIUM`, `HIGH`, or `CRITICAL`.
+    - `short_code` is a short code for the check. This should be a short, descriptive name for the check, separating words with hyphens. You should omit provider/service from this.
+    - `recommended_action` is a recommended remediation action for the check. This should be a short, descriptive sentence describing what the user should do to resolve the issue.
+    - `input` tells trivy what inputs this check should be applied to. Cloud provider checks should always use the `selector` input, and should always use the `type` selector with `cloud`. Check targeting Kubernetes yaml can use `kubenetes`, RBAC can use `rbac`, and so on.
+    - `subtypes` aid the engine to determine if it should load this policy or not for scanning. This can aid with the performance of scanning, especially if you have a lot of checks but not all apply to the IaC that you are trying to scan.
+
+### Generating an ID
+
+If you plan to contribue your check back into the [trivy-policies](https://github.com/aquasecurity/trivy-policies) repository, it will require a valid ID. 
+
+Running `make id` in the trivy-policies repository will provide you with the next available _ID_ for your rule. The ID is used in the Rego check to identify it.
+
+## Writing Rego Rules
+
+At last, it's time to write your rule code! Rules are defined using _OPA Rego_. You can find a number of examples in the `checks/cloud` directory ([Link](https://github.com/aquasecurity/trivy-policies/tree/main/checks/cloud)). The [OPA documentation](https://www.openpolicyagent.org/docs/latest/policy-language/) is a great place to start learning Rego. You can also check out the [Rego Playground](https://play.openpolicyagent.org/) to experiment with Rego, and [join the OPA Slack](https://slack.openpolicyagent.org/).
+
+Create a new file in `checks/cloud` ([Link](https://github.com/aquasecurity/trivy-policies/tree/main/checks/cloud)) with the name of your rule. You should nest it in the existing directory structure as applicable. The package name should be in the format `builtin.PROVIDER.SERVICE.ID`, e.g. `builtin.aws.rds.aws0176`.
+
+Now you'll need to write the rule logic. This is the code that will be executed to detect the issue. You should define a rule named `deny` and place your code inside this. Every check in Trivy needs to have a `deny` rule.
+
+```rego
+deny[res] {
+	instance := input.aws.rds.instances[_]
+	instance.engine.value == ["postgres", "mysql"][_]
+	not instance.iamauthenabled.value
+	res := result.new("Instance does not have IAM Authentication enabled", instance.iamauthenabled)
+}
+```
+
+The rule should return a result, which can be created using `result.new` (this function does not need to be imported, it is defined internally and provided at runtime). The first argument is the message to display, and the second argument is the resource that the issue was detected on.
```

Comment:
add (about the second argument): "you can pass any rego variable that reference reference a field of the input document."